src/hooks/__test__/use-companies.test.js

```diff
@@ -0,0 +1,28 @@
+import useCompanies from 'hooks/use-companies';
+import { renderReactQueryHook } from 'hooks/__test__/hook-test-utils.js';
+import MockAdapter from 'axios-mock-adapter';
+import axios from 'service/api';
+import mockedCompanies from 'test/fixtures/companies';
+
+const mock = new MockAdapter(axios);
+
+const renderUseCompaniesHook = () => renderReactQueryHook(useCompanies);
+
+describe('useCompanies hook', () => {
+  beforeEach(() => {
+    mock.onGet('/companies').reply(200, mockedCompanies);
+  });
+
+  it('isLoading during the fetch', async () => {
+    const hook = renderUseCompaniesHook();
+    expect(hook.result.current.isLoading).toBe(true);
+    await hook.waitForNextUpdate();
+    expect(hook.result.current.isLoading).toBe(false);
```

Comment:
NIT: in order to keep the same pattern, you could destruct the `result` and `waitForNextUpdate` just like you did in the `useCompaniesSelectionMenuItem` hook test:
```suggestion
    const { result, waitForNextUpdate } = renderUseCompaniesHook();
    expect(result.current.isLoading).toBe(true);
    await waitForNextUpdate();
    expect(result.current.isLoading).toBe(false);
```

Not a blocker, but if you choose to change it here, I suggest changing the other test of this hook as well as the `useDiscountRequest` tests